core/new-gui/src/app/workspace/service/validation/validation-workflow.service.ts

```diff
@@ -0,0 +1,141 @@
+          const status = this.validateOperator(operatorID);
+          this.operatorValidationStream.next({status, operatorID});
+         }
+        );
+
+    this.workflowActionService.getTexeraGraph().getLinkAddStream()
+      .subscribe(value => {
+        let operatorID = value.source.operatorID;
+        let status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+        operatorID = value.target.operatorID;
+        status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+      });
+
+
+    this.workflowActionService.getTexeraGraph().getLinkDeleteStream()
+      .subscribe(value => {
+        let operatorID = value.deletedLink.source.operatorID;
+        let status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+        operatorID = value.deletedLink.target.operatorID;
+        status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+      });
+
+    this.workflowActionService.getTexeraGraph().getOperatorPropertyChangeStream()
+      .subscribe(value => {
+        const operatorID = value.operator.operatorID;
+        const status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+      });
+  }
+
+  /**
+   * Gets the observable for operator validation change event.
+   * Contains a boolean variable that indicates:
+   *  - the new status for the validation of operator
+   */
+  public getOperatorValidationStream(): Observable<{status: boolean, operatorID: string}> {
+    return this.operatorValidationStream.asObservable();
+  }
+
+  /**
+   * if all ports of the operator are connected and all properties are completed
+   * then the operator is valid.
+   */
+  public validateOperator(operatorID: string): boolean {
+    if (!this.isOperatorIsolated(operatorID) && this.isJsonSchemaValiad(operatorID)) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * This method is used to check whether all required properties of the operator have been completed.
+   * if completed correctly, the operator box is valid.
+   */
+  private isJsonSchemaValiad(operatorID: string): boolean {
+    const operator = this.workflowActionService.getTexeraGraph().getOperator(operatorID);
+    let operatorSchema;
+    let isValid;
+    const ajv = new Ajv ({schemaId: 'auto', allErrors: true});
+    ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));
+    if (operator !== undefined) {
+      operatorSchema = this.operatorSchemaList.find(schema => schema.operatorType === operator.operatorType);
+      if (operatorSchema !== undefined) {
+        isValid =  ajv.validate(operatorSchema.jsonSchema, operator.operatorProperties);
+      }
+    }
+
+    if (isValid) {
+      return true;
+    }
+    return false;
+  }
```

Comment:
For this method, rewrite the checks so that it is not checking `operator !== undefined`, but when `operator === undefined`, throw an error. Same applies to `operatorSchema`.

When defining the Ajv checker for json schema
`const ajv = new Ajv ({schemaId: 'auto', allErrors: true});
    ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));`

Put these lines to the constructor of this service, so that you only need to create this Ajv object once (avoid repetition).